packages/override-tools/src/CrossProcessLock.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import * as net from 'net';
+
+/**
+ * CrossProcessLock implements a non-polling cross-process asynchronus barrier.
+ *
+ * - The lock is acquired by creating a named pipe
+ *
+ * - If named pipe creation fails due to an existing pipe, the process
+ *   subscribes to the pipe, waiting asynchronusly until it dies.
+ *
+ * - Once the pipe dies, any subscribers will race to acquire the lock (create
+ *   the pipe server)
+ *
+ */
+export default class CrossProcessLock {
```

Comment:
>class [](start = 15, length = 5)

Can you elaborate why we need this? So what if things break if enlistment is not used in a single threaded manner? In DevMain there is an existing tool called bw.bat which does that for those who need.

Reply:
The JS build will run builds for multiple packages in parallel. E.g. RNW at the same time as react-native-win32. If these run at the same time, they try to accmanipulate the same local git repo at the same time and the world blows up.

Usage is in CLI.js